udata/routing.py

```diff
@@ -59,7 +60,10 @@ class ModelConverter(BaseConverter):
     model = None
 
     def to_python(self, value):
-        obj = self.model.objects(slug=value).first()
+        try:
+            obj = self.model.objects(slug=value).first()
+        except InvalidQueryError:  # If the model doesn't have a slug.
```

Comment:
It's a matter of [EAFP](https://docs.python.org/2/glossary.html#term-eafp) vs. [LBYL](https://docs.python.org/2/glossary.html#term-lbyl). In that particular case, the fact that a slug is not present is the exception. BUT, we lost the specificity of catching only the issue if the query is invalid because of the missing slug… no I'm in doubt :)